### PR TITLE
[REFACTOR] Improve Convention Collective notification

### PR DESCRIPTION
**Pull request by hacherix**
Related to https://github.com/annuaire-entreprises-data-gouv-fr/search-infra/issues/418

### CODE REVIEW

**Comment by hacherix** [comment]
Note: require migration to Airflow 2.9.3 before merging.

File: config.py
Comment (HAEKADI) [inline config.py:304]:
```suggestion
# Caution: DARES file is case sensitive or returns a 404
```

File: helpers/__init__.py
Comment (HAEKADI) [inline helpers/__init__.py:1]: We generally avoid using relative imports as they can cause path issues, especially in a containerised environment. This is particularly relevant in our DAG context, where the Airflow configuration resides in a separate repository.

File: helpers/notification.py
Comment (HAEKADI) [inline helpers/notification.py:27]: I typically use NumPy/SciPy style Docstring for this project, but we can switch to Google style during refactoring if you prefer. 

Though I do admit I must have slipped one a few ones 😅 

The key is to maintain consistency by sticking to one style throughout in my opinion. So we should decide on one.
What do you think ?

File: helpers/notification.py
Comment (HAEKADI) [inline helpers/notification.py:7]: Very cool :)

File: workflows/data_pipelines/metadata/cc/DAG-get-metadata-cc.py
Comment (HAEKADI) [inline workflows/data_pipelines/metadata/cc/DAG-get-metadata-cc.py:17]: Yees!!!

File: workflows/data_pipelines/metadata/cc/task_functions.py
Comment (HAEKADI) [inline workflows/data_pipelines/metadata/cc/task_functions.py:141]: Since this is optional, I might just not mention it.

File: helpers/__init__.py
Comment (HAEKADI) [inline helpers/__init__.py:5]: Is there a specific reason you're only exposing the Notification module ?

**Review by HAEKADI — APPROVED** [review]
Looks good to me :)
Just a few minor details!

File: helpers/notification.py
Comment (hacherix) [reply helpers/notification.py:27]: I wrote it like this out of habits but indeed let's stay consistent. I'll convert it to Numpy style, good point!

File: workflows/data_pipelines/metadata/cc/task_functions.py
Comment (hacherix) [reply workflows/data_pipelines/metadata/cc/task_functions.py:141]: `File` definition is a `class File(TypedDict)` so a dictionnary.
If we don't setup `content_type` it will raise an error if we do `file["content_type"]`.
Unfortunately `TypedDict` does not allow for default values.

I was thinking about making it a dataclass but I noticed is used everywhere so I felt like it could wait for the bigger refactoring. I was afraid of creating bugs to other DAGs haha.

File: helpers/__init__.py
Comment (hacherix) [reply helpers/__init__.py:5]: I only added this one because I worked on it :D But of course we could add all the helpers! Should I add them now?

File: helpers/notification.py
Comment (HAEKADI) [reply helpers/notification.py:27]: Should we keep it and merge to Google for all Docstring in the refactor since it's more concise ?

Comment (HAEKADI) [reply helpers/notification.py:27]: After discussion, we'll go the Google route !!

File: helpers/__init__.py
Comment (HAEKADI) [reply helpers/__init__.py:5]: Sounds good!!

**Pull request merged**
